File: Andriod/Demo/Demo/app/src/main/java/com/example/demo/MusicAdapter.java
package com.example.demo;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.demo.databinding.RowMusicBinding;

import java.util.List;

public class MusicAdapter extends RecyclerView.Adapter<MusicAdapter.ViewHolder> {

    public Context context;
    public List<Integer> list;
    public MusicAdapter(MainActivity mainActivity, List<Integer> list) {
        this.list=list;
        context=mainActivity;
    }

    @NonNull
    @Override
    public MusicAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater=LayoutInflater.from(context);
        View view=inflater.inflate(R.layout.row_music,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MusicAdapter.ViewHolder holder, int position) {
        holder.binding.tvSongName.setText(list.get(position));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        public RowMusicBinding binding;
        public ViewHolder(View view){
            super(view);
            binding= DataBindingUtil.bind(view);
        }
    }
}

File: Andriod/Ceryl/app/src/main/java/com/ceryl/ceryl/adapter/MusicAdapter.kt
package com.ceryl.ceryl.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.ceryl.ceryl.activity.MainActivity
import com.ceryl.ceryl.R
import com.ceryl.ceryl.databinding.RowMusicBinding

class MusicAdapter(var mainActivity: MainActivity, var arrayList: ArrayList<String>) : RecyclerView.Adapter<MusicAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MusicAdapter.ViewHolder {
        val view=LayoutInflater.from(mainActivity).inflate(R.layout.row_music,parent,false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return arrayList.size
    }

    override fun onBindViewHolder(holder: MusicAdapter.ViewHolder, position: Int) {
        holder.binding?.tvImageName?.setText(arrayList.get(position))
    }

    class ViewHolder(view:View) : RecyclerView.ViewHolder(view){
        val binding:RowMusicBinding?
        init {
            binding=DataBindingUtil.bind(itemView)
        }
    }
}